### DISCUSSION TITLE
Thumbnail help

### DISCUSSION THREAD

**Discussion by clownnut** [mriscoc/Ender3V2S1#1295] [Q&A]
![20240425_141154](https://github.com/mriscoc/Ender3V2S1/assets/20946286/c1e66a51-135d-4593-940a-563a0e93725e)
![20240425_141133](https://github.com/mriscoc/Ender3V2S1/assets/20946286/ec47e07e-c3a5-4375-91e0-b9b75299e032)
Hi, I have an Ender 3 v2.
I got the thumbnail working before on this printer, then I upgraded my PC, and I reinstalled Cura (5.60) redownloaded the script enabled it in the extension, and installed python. Now, no matter what, any new file won't be a thumbnail. Just show the square code. but all the old files show the thumbnail fine. i just don't know what I didn't do to cause this.
any help will be greatly appreciated.

**Comment by Koky05** [comment]
Did you try to run same postprocess scipt from command line? Maybe it is missing "Pillow" module [GCode Preview](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview):
"For be able to generate g-code with thumbnail it is necessary to install [Python 3](https://www.python.org/downloads/) and a script/plugin in your slicer. Prusa/Super Slicer also requires the Python Pillow library."

**Reply by clownnut** [reply]
yes, i add the script to postprocess. i did install pillow and python3.
![thumbnail](https://github.com/mriscoc/Ender3V2S1/assets/20946286/0fb6313e-3460-4cd2-b314-4aa06e0758d4)

**Comment by Koky05** [comment]
My question was if you try to run it from command prompt.
`c:\Program Files\Python\python.exe C:\Users\NameOfUser\AppData\Roaming\cura\scripts\professionalfirmware.py c:\users\licai\Gcode\some_model.gcode`

**Reply by clownnut** [reply]
python isnt install to program files location. it's default to C:\Users\LiCai\AppData\Roaming\Python

C:\Users\LiCai\AppData\Roaming\cura\5.6\scripts

c:\users\licai\Gcode\some_model.gcode does not exist.

**Reply by Koky05** [reply]
`c:\users\licai\Gcode\some_model.gcode` is address to your sliced gcode file.

**Comment by clownnut** [comment]
> `c:\users\licai\Gcode\some_model.gcode` is address to your sliced gcode file.

i use cura.

**Reply by Koky05** [reply]
And it not generate gcode files for printer?
I use two different slicers on as Standalone app on my comp, second as docker version for remote slicing from work. Both generates gcode which I can send to printer via OctoPrint or copy to SD card. GCode file is only input variable for postprocess script.

**Reply by clownnut** [reply]
i checked all the new file, the gcode were there for the thumbnail, but not display when I select the file to print, while the old files will display the thumbnail.
